pipeline_steps/common.groovy

```diff
@@ -278,13 +278,54 @@ def archive_artifacts(){
   } finally{
     // still worth trying to archiveArtifacts even if some part of
     // artifact collection failed.
-    archiveArtifacts(
-      allowEmptyArchive: true,
-      artifacts: "artifacts_${env.BUILD_TAG}.tar.bz2,artifact_index.txt"
-    )
+    stash(
+      name: "artifacts",
+      includes: "artifacts_${env.BUILD_TAG}.tar.bz2")
```

Comment:
It turns out that there was only one fix needed to get the rax module to work on the cloud  slave, and that was to install libssl-dev. It's been a while since I last tried, so I thought it was going to be more work than that. Also, since we're uploading directly from the cloud node now, there's less of a need to remove the old artifacts since most nodes are for single use, so I omitted that part in my update.